python/ray/job_config.py

```diff
@@ -67,6 +54,24 @@ def serialize(self):
         job_config = self.get_proto_job_config()
         return job_config.SerializeToString()
 
+    def set_runtime_env(self, runtime_env):
+        # Lazily import this to avoid circular dependencies.
+        import ray._private.runtime_env as runtime_support
+        if runtime_env:
+            # Remove working_dir from the dict here, since that needs to be
+            # uploaded to the GCS after the job starts.
+            without_dir = dict(runtime_env)
```

Comment:
shouldn't this deep copy?

Reply:
i think it's fine because it looks like we only do the copy to let us do the `del without_dir["workier_dir"]` operation, but I really just moved this code out of the constructor without changing it, so cc @iycheng